dev-lang/vala/vala-0.40.3.recipe

```diff
@@ -0,0 +1,106 @@
+SUMMARY="An object-oriented programming language that uses the GObject system"
+DESCRIPTION="Vala is a new programming language that aims to bring modern \
+programming language features to GNOME developers without imposing any \
+additional runtime requirements and without using a different ABI \
+compared to applications and libraries written in C."
+HOMEPAGE="https://wiki.gnome.org/Projects/Vala"
+COPYRIGHT="2006-2018 Jürg Billeter
+	2006-2018 Raffaele Sandrini"
+LICENSE="GNU LGPL v2.1"
+REVISION="1"
+SOURCE_URI="https://download.gnome.org/sources/vala/${portVersion%.*}/vala-$portVersion.tar.xz"
+CHECKSUM_SHA256="6d4f1f0b2edaa0d4aa96f72735a9845e6b1acf80a0a2ae494b5d43e07833119e"
+SOURCE_DIR="vala-$portVersion"
+PATCHES="vala-$portVersion.patchset"
+
+ARCHITECTURES="!x86_gcc2 ?x86 ?x86_64"
+SECONDARY_ARCHITECTURES="?x86"
+
+libVersion="0.0.0"
+libVersionCompat="$libVersion compat >= ${libVersion%%.*}"
+portVers="${portVersion%.*}"
+
+PROVIDES="
+	vala$secondaryArchSuffix = $portVersion
+	cmd:vala$secondaryArchSuffix
+	cmd:vala_$portVers$secondaryArchSuffix
+	cmd:vala_gen_introspect$secondaryArchSuffix
+	cmd:vala_gen_introspect_$portVers$secondaryArchSuffix
+	cmd:valac$secondaryArchSuffix
+	cmd:valac_$portVers$secondaryArchSuffix
+	cmd:valadoc$secondaryArchSuffix
+	cmd:valadoc_$portVers$secondaryArchSuffix
+	cmd:vapigen$secondaryArchSuffix
+	cmd:vapigen_$portVers$secondaryArchSuffix
+	lib:libvala_$portVers$secondaryArchSuffix = $libVersionCompat
+	lib:libvaladoc_$portVers$secondaryArchSuffix = $libVersionCompat
+	devel:libvala_$portVers$secondaryArchSuffix = $libVersionCompat
+	devel:libvaladoc_$portVers$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libcdt$secondaryArchSuffix
+	lib:libcgraph$secondaryArchSuffix
+	lib:libglib_2.0$secondaryArchSuffix
+	lib:libgvc$secondaryArchSuffix
+	lib:libintl$secondaryArchSuffix
+	"
+REPLACES="
+	vala${secondaryArchSuffix}_common
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libcdt$secondaryArchSuffix
+	devel:libcgraph$secondaryArchSuffix
+	devel:libglib_2.0$secondaryArchSuffix
+	devel:libgvc$secondaryArchSuffix
+	devel:libintl$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:aclocal
+	cmd:autoreconf
+	cmd:awk
+	cmd:bison
+	cmd:cmp
+	cmd:diff
+	cmd:find
+	cmd:flex
+	cmd:gcc$secondaryArchSuffix
+	cmd:libtoolize$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	"
+
+defineDebugInfoPackage vala$secondaryArchSuffix \
+	$binDir/vala \
+	$binDir/vala-$portVers \
+	$binDir/valac \
+	$binDir/valac-$portVers \
+	$binDir/valadoc \
+	$binDir/valadoc-$portVers \
+	$binDir/vapigen \
+	$binDir/vapigen-$portVers \
+	$libDir/libvala-$portVers.so.$libVersion \
+	$libDir/libvaladoc-$portVers.so.$libVersion
+
+BUILD()
+{
+	autoreconf -vfi
+	runConfigure ./configure
+	make $jobArgs
+}
+
+INSTALL()
+{
+	make install
+
+	find "$libDir" -name '*.la' -delete
+}
```

Comment:
Hmm, if we drop the
```diff
	prepareInstalledDevelLibs libvala-$portVers libvaladoc-$portVers
	fixPkgconfig
```
then I think we also need to drop the `devel:lib` lines in `PROVIDES`.
Reminder: to check this you need to pass `-S` to haikuporter.

Reply:
Well we do need both lines, as our pkgconfig and linker is configured to search within `B_SYSTEM_DEVELOP_LIB_DIRECTORY`

Reply:
Yep, I agree.

(BTW, in [this comment](https://github.com/haikuports/haikuports/pull/2256#discussion_r180072230) I had only suggested the removal of the `packageEntries devel $developDir`.)

So, @Begasus, don't forget to restore the 
```
	prepareInstalledDevelLibs libvala-$portVers libvaladoc-$portVers
	fixPkgconfig
```
Thanks!

Reply:
Just a notice, I always use -S with haikuporter :)

Reply:
I probably won't hurt to include it anyway :) (running fixPkgconfig to make sure everything is correct) :)